tools/clang/unittests/HLSL/LinkerTest.cpp

```diff
@@ -526,6 +526,11 @@ TEST_F(LinkerTest, RunLinkMatArrayParam) {
   Link(L"main", L"ps_6_0", pLinker, {libName, libName2},
        {"alloca [24 x float]", "getelementptr [12 x float], [12 x float]*"},
        {});
+
+  Link(L"main", L"ps_6_9", pLinker, {libName, libName2},
+       {"alloca [2 x <12 x float>]",
+        "getelementptr [12 x float], [12 x float]*"},
```

Comment:
I'm not sure how to tell if this is right.  Shouldn't this be dealing with a vector rather than an array here?  Can I see the IR?

Reply:
Well I was ready to write something here about how this was compiled as 6.3 because that's what the file says, but it's actually 6_x. The 12 element array comes from HLMatrixLower, which I hadn't touched.

Reply:
Okay, it's just scratch space used to move the elements of the matrix around. They use vectors too. I don't know why the test cares about it so much, but it doesn't need changing. I'll take a look at leveraging long vectors for matrices later, but this works perfectly fine for now.